File: language/runtime/apple/core/Tests/RuntimeHostAppleCoreTests/Fixtures/LifecycleFixtures.swift
import RuntimeHostAppleCore

/// One recording lifecycle sink for Apple core tests.
@MainActor
final class RecordingLifecycleSink: LifecycleEvents {
  var events: [RuntimeHostLifecycleEvent] = []

  func sendLifecycleEvent(_ event: RuntimeHostLifecycleEvent) {
    events.append(event)
  }
}

/// Build one lifecycle event fixture.
func lifecycleEvent(
  _ sourceKind: RuntimeHostLifecycleSourceKind,
  _ state: RuntimeHostLifecycleState
) -> RuntimeHostLifecycleEvent {
  RuntimeHostLifecycleEvent(
    sourceKind: sourceKind,
    state: state
  )
}

File: language/runtime/android/kotlin/src/test/kotlin/dev/destack/runtime/android/module/lifecycle/LifecycleFixtures.kt
package dev.destack.runtime.android

import dev.destack.runtime.android.module.lifecycle.LifecycleEvents
import dev.destack.runtime.android.module.lifecycle.RuntimeHostLifecycleEvent
import dev.destack.runtime.android.module.lifecycle.RuntimeHostLifecycleSourceKind
import dev.destack.runtime.android.module.lifecycle.RuntimeHostLifecycleState

/**
 * One recording lifecycle sink for Android host tests.
 */
internal class RecordingLifecycleSink : LifecycleEvents {
    val events: MutableList<RuntimeHostLifecycleEvent> = mutableListOf()

    override fun sendLifecycleEvent(
        event: RuntimeHostLifecycleEvent,
    ) {
        events += event
    }
}

/**
 * Build one application lifecycle event fixture.
 */
internal fun applicationLifecycleEvent(
    state: RuntimeHostLifecycleState,
): RuntimeHostLifecycleEvent {
    return RuntimeHostLifecycleEvent(
        sourceKind = RuntimeHostLifecycleSourceKind.Application,
        state = state,
    )
}

/**
 * Build one activity lifecycle event fixture.
 */
internal fun activityLifecycleEvent(
    state: RuntimeHostLifecycleState,
): RuntimeHostLifecycleEvent {
    return RuntimeHostLifecycleEvent(
        sourceKind = RuntimeHostLifecycleSourceKind.Activity,
        state = state,
    )
}
